tools/rustfmt/srcs/main.rs

```diff
@@ -71,74 +69,111 @@ fn query_rustfmt_targets(options: &Config) -> Vec<String> {
         .collect()
 }
 
-/// Build a list of Bazel targets using the `rustfmt_aspect` to produce the
-/// arguments to use when formatting the sources of those targets.
-fn generate_rustfmt_target_manifests(options: &Config, targets: &[String]) {
-    let build_args = vec![
-        "build".to_owned(),
+/// Query for all `*.rs` files in a workspace that are dependencies of targets with an unspecified edition.
+/// This indicates that the edition is expected to be provided by the `rust_toolchain`.
+fn unspecified_edition_query(bazel_bin: &Path, scope: &str, current_dir: &Path) -> Vec<String> {
+    let query_args = vec![
+        "query".to_owned(),
+        // Query explanation:
+        // Filter all local targets ending in `*.rs`.
+        //     Get all source files.
+        //         Get direct dependencies.
+        //             Get all targets with an empty `edition` attribute.
+        //             Except for targets tagged with `norustfmt`.
+        //             And except for targets with a populated `crate` attribute since `crate` defines edition for this target
         format!(
-            "--aspects={}//rust:defs.bzl%rustfmt_aspect",
-            env!("ASPECT_REPOSITORY")
+            r#"let scope = set({scope}) in filter("^//.*\.rs$", kind("source file", deps(attr(edition, "{edition}", $scope) except attr(tags, "(^\[|, )norustfmt(, |\]$)", $scope) + attr(crate, ".*", $scope), 1)))"#,
```

Comment:
It feels worth factoring out this query from the two places it's present, because it's pretty intricate and I imagine coordinating changes across the two copies will be hard?

Reply:
Updated, thanks for calling this out!